Documentation/templates/strings/structured-swift4.md

```diff
@@ -25,10 +25,10 @@ You can customize some elements of this template by overriding the following par
 | Parameter Name | Default Value | Description |
 | -------------- | ------------- | ----------- |
 | `enumName` | `L10n` | Allows you to change the name of the generated `enum` containing all string tables. |
+| `forceFileNameEnum` | N/A | Setting this parameter will generate an `enum <FileName>` _even if_ only one FileName was provided as input. |
+| `lookupFunction` | N/A | Allows you to set your own custom localization function. The function needs to have as signature: `(key: String, table: String) -> String`. If the function has named parameters, you must provide the complete function signature, including those named parameters – e.g. `yourFunctionName(forKey:table:)`. |
```

Comment:
A bit confusing imho that you both say "signature is `(key: String, table: String)`" and also "if function has named parameters. It's not clear if the parameters need to explicitly have the exact `key:` and `table:` parameter names or not, as those two sentences seems a bit contradictory, one suggesting they params must be named that way and the other suggesting that the function can decide to not have named params…

```suggestion
| `lookupFunction` | N/A | Allows you to set your own custom localization function. The function expects the key and table as parameters (in that order) and needs to have as signature: `(String, String) -> String`. If the function has named parameters, you must provide the complete function signature, including those named parameters – e.g. `yourFunctionName(forKey:table:)`. |
```

Reply:
Reason I changed the wording like that is to give it a clear meaning. They're both `String`s, but the first one is the localization key, the 2nd one is the table key. It also unifies the documentation for other `lookupFunction`s (in #738), where we can have signatures like `(name: String, family: String, path: String) -> URL?` for the fonts templates.

Reply:
Right. Then maybe mention that named labels for parameters are optional?

```suggestion
| `lookupFunction` | N/A | Allows you to set your own custom localization function. The function needs to have as signature: `(key: String, table: String) -> String`. The parameters of your function can have any name (or even no external name), but if it has named parameters, you must provide the complete function signature, including those named parameters – e.g. `yourFunctionName(forKey:table:)`. |
```